test/MicrosoftTeams.spec.ts

```diff
@@ -169,20 +181,63 @@ describe("MicrosoftTeams", () => {
     let getContextMessage = findMessageByFunc("getContext");
     expect(getContextMessage).not.toBeNull();
 
-    processMessage({
-      origin: "https://some-malicious-site.com/",
-      source: microsoftTeams._window.parent,
-      data: {
-        id: getContextMessage.id,
-        args: [
-          {
-            groupId: "someMaliciousValue"
-          }
-        ]
-      } as MessageResponse
-    } as MessageEvent);
+    unSupportedDomains.forEach(unSupportedDomain => {
+      processMessage({
+        origin: unSupportedDomain,
+        source: microsoftTeams._window.parent,
+        data: {
+          id: getContextMessage.id,
+          args: [
+            {
+              groupId: "someMaliciousValue"
+            }
+          ]
+        } as MessageResponse
+      } as MessageEvent);
+
+      expect(callbackCalled).toBe(false);
+    });
+  });
+
+  it("should allow messages from supported domains", () => {
+    initializeWithContext("content");
+    const supportedDomains = [
+      "https://teams.microsoft.com",
+      "https://teams.microsoft.us",
+      "https://int.teams.microsoft.com",
+      "https://devspaces.skype.com",
+      "http://dev.local",
+      "https://microsoft.sharepoint.com",
+      "https://a.b.sharepoint.com",
+      "https://a.b.c.sharepoint.com",
+      "https://msft.spoppe.com",
+      "https://microsoft.sharepoint-df.com"
+    ];
+
+    let callbackCalled = false;
+    microsoftTeams.getContext(() => {
+      callbackCalled = true;
+    });
 
-    expect(callbackCalled).toBe(false);
+    let getContextMessage = findMessageByFunc("getContext");
+    expect(getContextMessage).not.toBeNull();
+
+    supportedDomains.forEach(supportedDomain => {
+      processMessage({
+        origin: supportedDomain,
+        source: microsoftTeams._window.parent,
+        data: {
+          id: getContextMessage.id,
+          args: [
+            {
+              groupId: "someMaliciousValue"
+            }
+          ]
+        } as MessageResponse
+      } as MessageEvent);
+
+      expect(callbackCalled).toBe(true);
```

Comment:
What resets callbackCalled to false for each item in supportedDomains? I think this ends up checking only that the callback was called for at least *one* of the domains, not each one.

Reply:
thanks i am fixing Unit tests logic